source/credentials_provider_imds.c

```diff
@@ -565,86 +351,403 @@ static int s_make_imds_http_query(
         .request = request,
     };
 
-    struct aws_http_stream *stream =
-        impl->function_table->aws_http_connection_make_request(imds_user_data->connection, &request_options);
+    imds_user_data->query_state += 1;
```

Comment:
This works but I can see it blowing up without us knowing due to future change in the enum.  At bare minimum we should document the state enum and every place we modify it arithmetically with warnings/disclaimers about changing the state enum.  Alternatively, we could add advance/rewind functions for the state enum that operate on the specific enum values.

Reply:
this was actually a bug waiting to happen. The state functions are responsible for state changes. We just lucked out that I was setting it after the call to an explicit value.

Reply:
deleted